File: Philadelphia Workshop/4 - GraphEQ/Java_Scaffold/GraphEQ.java

import bridges.base.*;
import bridges.connect.Bridges;
import bridges.data_src_dependent.EarthquakeUSGS;

import java.util.Comparator;
import java.util.List;

public class GraphEQ {

    public static double calcDistance(double la1, double lo1, double la2, double lo2) {
        final int R = 6371; // Radius of the earth in KM
        // Haversine formula to calculate a value between 0 and 1 between 2 points on a sphere, 1 being the
        // opposite side of the sphere
        double laDistance = Math.toRadians(la2 - la1);
        double loDistance = Math.toRadians(lo2 - lo1);
        double a = Math.sin(laDistance / 2) * Math.sin(laDistance / 2)
                + Math.cos(Math.toRadians(la1)) * Math.cos(Math.toRadians(la2))
                * Math.sin(loDistance / 2) * Math.sin(loDistance / 2);
        double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));

        double distance = R * c;    //convert to KM
        return distance;
    }

    public static void main(String[] args) throws Exception {
        Bridges bridges = new Bridges(4, "bridges_workshop", "1298385986627");

		/*	TODO:
		 *	Initialize a Graph of your choice
		 *	Grab Earthquake data and store it in a List
		 *	Sort that list in decending order
		 */

        for (int i = 0; i < 100; ++i) {
		/* TODO:
		 * Add the top 99 Earthquakes to the graph
		 * Set each earthquake's location based on its latitude and longitude
		 * ex: graph.getVisualizer(key).setLocation(earthquake.getLongit(), earthquake.getLatit());
		 * Tweak the colors or other visual elements if you wish
		 */
        }

        bridges.setCoordSystemType("equirectangular");
        bridges.setDataStructure(graph);
        bridges.setMapOverlay(true);
        bridges.setServer("clone");
        bridges.setTitle("Earthquake Map");
        bridges.visualize();

        for (int i = 0; i < 100; ++i) {
		/* TODO:
		 * Compare the distances between all vertexes in the graph, drawing an edge 
		 *	if they are within some threshold. A method is provided to give a rough
		 *	estimate between 2 lat,long points.
		 *
		 * example usage: calcDistance(eq1.getLatit(), eq1.getLongit(),
		 *							   eq2.getLatit(), eq2.getLongit());
		 * which returns a double representing the distance of two points in KM
		 */
        }

        bridges.visualize();

        for (int i = 0; i < 100; ++i) {
		/* TODO:
		 * Reset the locations of the vertices by setting their location to
		 *	Double.POSITIVE_INFINITY
		 *
		 *	ex: graph.getVisualizer(key).setLocation(Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY) 
		 */
        }

        bridges.setMapOverlay(false);
        bridges.visualize();
    }
}

File: Philadelphia Workshop/4 - GraphEQ/C++_Scaffold/GraphEQ.cpp
#include "Bridges.h"
#include "DataSource.h"
#include "Array.h"
#include "SLelement.h"
#include "GraphAdjList.h"

using namespace bridges;

double degree_to_rad(double d) {
  return d*M_PI/180;
}

double calcDistance(double la1, double lo1, double la2, double lo2) {
  double R = 6371; // Radius of the earth in km
  // Haversine formula to calculate a value between 0 and 1 between 2 points on a sphere, 1 being the
  // opposite side of the sphere
  double laDistance = degree_to_rad(la2 - la1);
  double loDistance = degree_to_rad(lo2 - lo1);
  double a = sin(laDistance / 2) * sin(laDistance / 2)
    + cos(degree_to_rad(la1)) * cos(degree_to_rad(la2))
    * sin(loDistance / 2) * sin(loDistance / 2);

  double c = 2. * atan2(sqrt(a), sqrt(1 - a));

  
  double distance = R * c;    //convert to km
  return distance;
}


int main() {

	//create the Bridges object, set credentials
	Bridges::initialize(104, "bridges_workshop", "1298385986627");
	
	/* TODO:
	 * Grab 10000 earthquakes.
	 * Retain only the 100 earthquakes of highest magnitude.
	 */
	
	/* TODO:
	 * Initialize a Graph of your choice (recommend a GraphAdjList <int, EarthquakeUSGS>) 
	 * Tell Bridges to visualize that graph.
	 * Add the Earthquakes to the graph as individual vertices.
	 * Set each earthquake's location based on its longitude and latitude (Use ElementVisualizer::setLocation() ).
	 * Tweak the colors or other visual elements if you wish.
	 */
	
	
	Bridges::setCoordSystemType("equirectangular");
	Bridges::setMapOverlay(true);

	// visualize the list
	Bridges::visualize();
	
	/* TODO: 
	 * 
	 * Compare the distances between all vertexes in the graph,
	 * drawing an edge if they are within 500km.
	 *
	 * Use calcDistance to compute the distance between two
	 * earthquakes.
	 */
	
	
	// visualize the list
	Bridges::visualize();

	// Visualize the graph just by itself
	
	/* TODO:
	 *
	 * Reset the locations of the vertices by setting their location to INFINITY
	 */


	Bridges::setMapOverlay(false);

	
	// visualize the list
	Bridges::visualize();

	
	return 0;
}
